auth-api/src/auth_api/services/affiliation_mapping.py

```diff
@@ -0,0 +1,121 @@
+# Copyright © 2025 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Service for managing Affiliation Mapping data."""
+import datetime
+from sqlalchemy import and_, or_
+import re
+from dataclasses import asdict
+from typing import Dict, List, Optional
+
+from flask import current_app
+from requests.exceptions import HTTPError
+from sqlalchemy.orm import contains_eager, subqueryload
+from structured_logging import StructuredLogging
+
+from auth_api.exceptions.errors import Error
+from auth_api.models import db
+from auth_api.models.affiliation import Affiliation as AffiliationModel
+from auth_api.models.entity import Entity
+from auth_api.models.affiliation_mapping import AffiliationMapping
+from auth_api.models.dataclass import AffiliationSearchDetails
+
+logger = StructuredLogging.get_logger()
+
+
+class AffiliationMappingService:  # pylint: disable=too-few-public-methods
+    """Manages all aspect of Affiliation Mapping data.
+
+    This manages updating, retrieving Affiliations in Affilaition Mapping data via the AffiliationMapping model.
+    """
+
+    def __init__(self, model):
+        """Return an AffiliationMapping Service."""
+        self._model = model
+
+    def get_filtered_affiliations(org_id: int, search_details: AffiliationSearchDetails) -> List[AffiliationModel]:
+        """Get affiliations from DB based on priority mapping logic.
+        - If no filters are applied (initial load), show only top 5 pages max.
+        - If filters are applied, allow full pagination.
+        """
+        # Determine pagination settings
+        limit = search_details.limit
+        page = search_details.page
+        MAX_INITIAL_PAGES = 5
+
+        # Check if it's an unfiltered "initial load"
```

Comment:
rename variable to incorporate comment if it's important?
`is_unfiltered_initial_load`?